File: wk1/apaxianparent.java
// problem link: https://open.kattis.com/problems/apaxianparent

import java.util.*;
import java.io.*;
public class apaxianparent {
    public static void main(String[] args) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(System.in));
        String[] yp = r.readLine().split(" ");
        if("aeiou".indexOf(yp[0].charAt(yp[0].length() - 1)) > -1)
            System.out.println(yp[0].substring(0, yp[0].length() - 1) + "ex" + yp[1]);
        else if(yp[0].charAt(yp[0].length() - 2) == 'e' && yp[0].charAt(yp[0].length() - 1) == 'x')
            System.out.println(yp[0] + yp[1]);
        else
            System.out.println(yp[0] + "ex" + yp[1]);
    }
}

File: wk3/cpp/apaxianparent.cpp
// problem link: https://open.kattis.com/problems/apaxianparent

#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios::sync_with_stdio(false); cin.tie(0); cout.tie(0);
    
    string y, p, v = "aeiou";
    cin >> y >> p;
    if(v.find(y[y.size() - 1]) != string::npos)
        cout << y.substr(0, y.size() - 1) << "ex" << p;
    else if(y[y.size() - 2] == 'e' && y[y.size() - 1] == 'x')
        cout << y << p;
    else
        cout << y << "ex" << p;
    
    return 0;
}